R/type-data-frame.R

```diff
@@ -83,7 +83,7 @@ vec_type2.data.frame.data.frame <- function(x, y, ...) df_col_type2(x, y, ...)
 #' @method vec_type2.data.frame default
 #' @export
 vec_type2.data.frame.default <- function(x, y, ..., x_arg = "", y_arg = "") {
-  stop_incompatible_type(x, y, x_arg = x_arg, y_arg = y_arg)
+  vec_default_type2(x, y, x_arg = x_arg, y_arg = y_arg)
```

Comment:
You should make a note to deprecate the `stop_` functions

Reply:
I thought they might still be useful in case unspecified needs to be treated specially by the method (I'm not sure if that will be ever needed though).